onboarding/source/amb_light_service.c

```diff
@@ -33,21 +53,82 @@ static void lightServiceTask(void * pvParameters);
 
 obc_error_code_t initLightService(void) {
     /* USER CODE BEGIN */
-    // Create the task and queue here. Return error code if task/queue was not created successfully.
+    if (lightServiceHandle == NULL) {
+        lightServiceHandle = xTaskCreateStatic(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                lightServiceStack,
+                                &lightServiceBuffer);
+                                    }
+    if (lightServiceHandle == NULL)
+    {
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    }
 
-    /* USER CODE END */
+    if (lightserviceQueue == NULL) {
+        lightserviceQueue = xQueueCreateStatic(LIGHT_QUEUE_LENGTH, LIGHT_QUEUE_SIZE,uclightQueueStorageArea, &xlightStaticQueue);
+    }
+    if (lightserviceQueue == NULL) 
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
+    
+    
+return OBC_ERR_CODE_SUCCESS;     
+}                      
+/* USER CODE END */
+
+
+uint16_t getLightSensorData(void)
```

Comment:
Btw nice job making a helper function for this. This makes it much easier to debug/test code.